pages/api/v1/answers.ts

```diff
@@ -562,101 +568,110 @@ async function Angela( // Good ol' Angela
 
-          return { nonSystemMessages, cost: totalCost, numUserQueries };
-        }
+              return { nonSystemMessages, cost: totalCost, numUserQueries };
+            }
 
-        const actionToHttpRequest: ActionToHttpRequest = {
-          action: chosenAction,
-          parameters: command.args,
-          organization: org,
-          stream: streamInfo,
-          userApiKey: reqData.user_api_key ?? "",
-        };
-
-        if (
-          ["get", "head", "options", "connect"].includes(
-            chosenAction.request_method!.toLowerCase()
-          )
-        ) {
-          const { url, requestOptions } =
-            constructHttpRequest(actionToHttpRequest);
-          let out;
-          try {
-            out = await makeHttpRequest(url, requestOptions, currentHost);
-            out = processAPIoutput(out, chosenAction);
-          } catch (e) {
-            console.error(e);
-            // @ts-ignore
-            out = `Failed to call ${url}\n\n${e.toString()}`;
-          }
-          console.log("Output from API call:", out);
-          const outMessage = {
-            role: "function",
-            name: command.name,
-            content: JSON.stringify(out, null, 2),
-          } as ChatGPTMessage;
-          streamInfo(outMessage);
-          nonSystemMessages.push(outMessage);
-        } else {
-          toConfirm.push({
-            actionId: chosenAction.id,
-            args: command.args,
-            name: command.name,
-          });
+            const actionToHttpRequest: ActionToHttpRequest = {
+              action: chosenAction,
+              parameters: command.args,
+              organization: org,
+              stream: streamInfo,
+              userApiKey: reqData.user_api_key ?? "",
+            };
 
-          awaitingConfirmation = true;
-        }
-      }
+            if (
+              ["get", "head", "options", "connect"].includes(
+                chosenAction.request_method!.toLowerCase()
+              )
+            ) {
+              const { url, requestOptions } =
+                constructHttpRequest(actionToHttpRequest);
+              let out;
+              try {
+                out = await makeHttpRequest(url, requestOptions, currentHost);
+                out = processAPIoutput(out, chosenAction);
+              } catch (e) {
+                console.error(e);
+                // @ts-ignore
+                out = `Failed to call ${url}\n\n${e.toString()}`;
+              }
+              console.log("Output from API call:", out);
+              const outMessage: GPTMessageInclSummary = {
+                role: "function",
+                name: command.name,
+                content:
+                  typeof out === "string" ? out : JSON.stringify(out, null, 2),
+              };
+
+              // If >1500 tokens, summarise the message
+              if (
+                typeof out === "string" &&
+                getTokenCount([outMessage]) > 1500
```

Comment:
this threshold seems quite high, is a 1000 word message not worth summarising? that's like 3 pages of a book

Reply:
Good point, will cut it to 500.

I expected the summarisation to be slower and worse than it is, so wanted it to have all the information unless a response was super long